ash/src/extensions/khr/video_encode_queue.rs

```diff
@@ -0,0 +1,84 @@
+//! <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_KHR_video_encode_queue.html>
+
+use core::mem;
+use core::ptr;
+
+use crate::prelude::*;
+use crate::vk;
+use crate::RawMutPtr;
+
+impl crate::khr::video_encode_queue::Device {
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdEncodeVideoKHR.html>
+    #[inline]
+    #[doc(alias = "vkCmdEncodeVideoKHR")]
+    pub unsafe fn cmd_encode_video(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        encode_info: &vk::VideoEncodeInfoKHR<'_>,
+    ) {
+        (self.fp.cmd_encode_video_khr)(command_buffer, encode_info)
+    }
+
+    /// Retrieves the length of the byte slice to pass to [`get_encoded_video_session_parameters`][Self::get_encoded_video_session_parameters].
+    pub unsafe fn get_encoded_video_session_parameters_len(
+        &self,
+        session_parameters_info: &vk::VideoEncodeSessionParametersGetInfoKHR<'_>,
+        feedback_info: Option<&mut vk::VideoEncodeSessionParametersFeedbackInfoKHR<'_>>,
+    ) -> VkResult<usize> {
+        let mut len = mem::MaybeUninit::uninit();
+        (self.fp.get_encoded_video_session_parameters_khr)(
+            self.handle,
+            session_parameters_info,
+            feedback_info.as_raw_mut_ptr(),
+            len.as_mut_ptr(),
+            ptr::null_mut(),
+        )
+        .assume_init_on_success(len)
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetEncodedVideoSessionParametersKHR.html>
+    ///
+    /// Call [`get_encoded_video_session_parameters_len`][Self::get_encoded_video_session_parameters] to determine the correct length of the output slice.
+    #[inline]
+    #[doc(alias = "vkGetEncodedVideoSessionParametersKHR")]
+    pub unsafe fn get_encoded_video_session_parameters(
+        &self,
+        session_parameters_info: &vk::VideoEncodeSessionParametersGetInfoKHR<'_>,
+        feedback_info: Option<&mut vk::VideoEncodeSessionParametersFeedbackInfoKHR<'_>>,
+        out: &mut [u8],
```

Comment:
My gripe with this API was forcing the caller to initialize the byte-slice. `&mut [MaybeUninit<u8>]` seems more fitting, and it allows callers to pass [`Vec::spare_capacity_mut()`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.spare_capacity_mut) directly after reserving enough capacity.

Reply:
Looks like there's also a new helper for that: [MaybeUninit::uninit_array](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.uninit_array)

Reply:
I wouldn't call it new: I've been waiting _for years_ for those helpers to stabilize 😅. Likewise for `slice_assume_init_*()`.